src/lib/EventListener/InteractiveLoginListener.php

```diff
@@ -37,8 +48,47 @@ public static function getSubscribedEvents()
 
     public function onInteractiveLogin(InteractiveLoginEvent $event)
     {
-        // This loads a generic user and assigns it back to the event.
-        // You may want to create users here, or even load predefined users depending on your own rules.
-        $event->setApiUser($this->userService->loadUserByLogin('lolautruche'));
+	    $userService = $this->repository->getUserService();
+        $username = $event->getAuthenticationToken()->getUsername();
+
+        try {
+	        $event->setApiUser($userService->loadUserByLogin($username));
+        } catch (\eZ\Publish\API\Repository\Exceptions\NotFoundException $exception) {
+        	$searchDn       = $this->container->getParameter("ldap_auth_search_dn");
+        	$password       = $this->container->getParameter("ldap_auth_search_password");
+        	$queryString    = $this->container->getParameter("ldap_auth_query_string");
+        	$baseDn         = $this->container->getParameter("ldap_auth_base_dn");
+        	$targetGroup    = $this->container->getParameter("ldap_auth_target_usergroup");
+
+        	// Login to LDAP server and get user attributes
+        	$this->ldap->bind($searchDn, $password);
+        	$query = $this->ldap->query($baseDn, str_replace("{username}", $username, $queryString));
+	        $results = $query->execute();
+
+	        // Prepare user details
+	        $ldapUser = $results->toArray()[0]->getAttributes();
+	        $email = $ldapUser["mail"][0] ?? "email@example.com";
+	        $password = $ldapUser["userPassword"][0];
```

Comment:
You should probably use `PropertyAccess` component to access `$ldapUser` details, since you can't guarantee that the keys you're trying to access are available.

Reply:
Yes, maybe